src/save/encode.ts

```diff
@@ -29,13 +34,19 @@ const defaultPng: EncodeOptionsPng = { format: 'png' };
  * @returns The encoded image.
  */
 export function encode(
-  image: Image,
-  options: EncodeOptionsPng | EncodeOptionsJpeg = defaultPng,
+  image: Image | Mask,
+  options: EncodeOptionsBmp | EncodeOptionsPng | EncodeOptionsJpeg = defaultPng,
 ): Uint8Array {
-  if (options.format === 'png') {
-    return encodePng(image, options.encoderOptions);
-  } else if (options.format === 'jpg' || options.format === 'jpeg') {
-    return encodeJpeg(image, options.encoderOptions);
+  if (image instanceof Image) {
+    if (options.format === 'png') {
+      return encodePng(image, options.encoderOptions);
+    } else if (options.format === 'jpg' || options.format === 'jpeg') {
+      return encodeJpeg(image, options.encoderOptions);
+    } else {
+      throw new RangeError(`invalid format: ${options.format}`);
+    }
+  } else if (options.format === 'bmp') {
+    return encodeBmp(image);
```

Comment:
encodeBmp should accept both image and Mask (non-binary images can be stored in the bmp format), but should throw if an image is passed with a message that it is not implemented.